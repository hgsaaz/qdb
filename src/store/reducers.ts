/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from '@reduxjs/toolkit';

import { InjectedReducersType } from 'utils/types/injector-typings';
import { RootState } from 'types/RootState';

export const initialReducers: InjectedReducersType = Object.keys(
  {} as RootState,
).reduce((result, reducerKey) => {
  // Create empty reducers for keys that don't have loaded dynamic reducer yet
  // They will be replaced by the real one
  // See https://github.com/rt2zz/redux-persist/pull/1047/files
  result[reducerKey] = (state = null) => state;
  return result;
}, {});

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export function createReducer(injectedReducers: InjectedReducersType = {}) {
  // Initially we don't have any injectedReducers, so returning identity function to avoid the error
  if (Object.keys(injectedReducers).length === 0) {
    return state => state;
  } else {
    return combineReducers({
      ...initialReducers,
      ...injectedReducers,
    });
  }
}
