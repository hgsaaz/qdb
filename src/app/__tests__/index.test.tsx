import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { App } from '../index';
import { HelmetProvider } from 'react-helmet-async';
import { Store } from '@reduxjs/toolkit';
import { configureAppStore } from 'store/configureStore';
import { ThemeProvider } from 'styles/theme/ThemeProvider';
const store = configureAppStore();

const renderAppPage = (store: Store) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <HelmetProvider>
          <App />
        </HelmetProvider>
      </ThemeProvider>
    </Provider>,
  );

describe('<App />', () => {
  beforeAll(() => {
    // mocking matchMedia
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  let component: ReturnType<typeof renderAppPage>;
  beforeEach(() => {
    component = renderAppPage(store);
    expect(store.getState().user.username).toEqual('1');
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render and match the snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
