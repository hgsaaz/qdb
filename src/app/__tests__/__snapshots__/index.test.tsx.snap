// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<BrowserRouter>
  <Helmet
    defaultTitle="QDB"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="QDB"
  >
    <meta
      content="QDB application"
      name="description"
    />
  </Helmet>
  <withRouter(PageLayout) />
  <UNDEFINED />
</BrowserRouter>
`;
